apps/qubit/modules/object/actions/exportAction.class.php

```diff
@@ -0,0 +1,166 @@
+  }
+
+  protected function addField($name)
+  {
+    switch ($name)
+    {
+      case 'levels':
+        $this->form->setValidator('levels', new sfValidatorPass);
+
+        foreach (QubitTerm::getLevelsOfDescription() as $item)
+        {
+          $this->choices[$item->id] = $item->__toString();
+        }
+
+        $size = count($this->choices);
+        if ($size === 0)
+        {
+          $size = 4;
+        }
+
+        $this->form->setWidget('levels', new sfWidgetFormSelect(array('choices' => $this->choices, 'multiple' => true), array('size' => $size)));
+
+        break;
+
+      default:
+        return parent::addField($name);
+    }
+  }
+
+  protected function processField($field)
+  {
+    switch ($field->getName())
+    {
+      case 'levels':
+        $this->levels = $this->form->getValue('levels');
+        if (empty($this->levels))
+        {
+          $this->levels = array();
+        }
+
+        break;
+    }
+  }
+
+  protected function doBackgroundExport($request)
+  {
+    // Export type, CSV or XML?
+    $exportType = $request->getParameter('format', 'xml');
+
+    // We will use this later to redirect users back to the export page
+    if (isset($this->getRoute()->resource))
+    {
+      $exportRoute = array($this->getRoute()->resource, 'module' => 'object', 'action' => 'export', 'type' => $exportType);
+    }
+    else
+    {
+      $exportRoute = array('module' => 'object', 'action' => 'export', 'type' => $exportType);
+    }
+
+    // Create array of selections to pass to background job where Term ID will
+    // be key, and Term description is value.
+    foreach($this->levels as $key => $value)
+    {
+      $levelsOfDescription[$value] = $this->choices[$value];
+    }
+
+    $options = array('params' => array('fromClipboard' => true,
+                                       'slugs' => $this->context->user->getClipboard()->getAll()),
+                     'include-all-levels' => ('on' == $request->getParameter('includeAllLevels')) ? true : false,
+                     'current-level-only' => ('on' == $request->getParameter('includeDescendants')) ? false : true,
+                     'public' => ($request->getParameter('includeDrafts') == 'on') ? false : true,
+                     'objectType' => $request->getParameter('objectType'),
+                     'levels' => $levelsOfDescription);
+
+    try
+    {
+      if ('CSV' == strtoupper($exportType))
+      {
+        QubitJob::runJob('arInformationObjectCsvExportJob', $options);
+      }
+      else
+      {
+        QubitJob::runJob('arXmlExportJob', $options);
+      }
+
+      // Let user know export has started
+      sfContext::getInstance()->getConfiguration()->loadHelpers(array('Url'));
+      $jobManageUrl = url_for(array('module' => 'jobs', 'action' => 'browse'));
+      $message = '<strong>Export of descriptions initiated.</strong> Check <a href="'. $jobManageUrl . '">job management</a> page to download the results when it has completed.';
```

Comment:
That's not going to work, the i18n extractor finds the strings statically. You need to put the string literal inside `__()` AFAIK.

If you need replacements, use:
```php
__('My name is %1%', array('%1%' => $name));
```

We try not to include HTML tags in our i18n strings:

```php
__('Export of descriptions initiated. %1%Check job management%2% page to download the results when it has completed.', array(
    '%1%' => sprintf('<a href="%s">', url_for(...)),
    '%2' => '</a>'));
```

I think the idea is to avoid having the translator writing HTML, but I'm not sure if AtoM is consistently doing that, probably not.